libs/sdk-ui-ext/src/internal/dashboardEmbedding/hooks/useDeleteWidgetAlert.ts

```diff
@@ -0,0 +1,76 @@
+// (C) 2020-2021 GoodData Corporation
+import { IAnalyticalBackend, IWidgetAlert } from "@gooddata/sdk-backend-spi";
+import {
+    GoodDataSdkError,
+    useBackend,
+    useCancelablePromise,
+    UseCancelablePromiseCallbacks,
+    UseCancelablePromiseState,
+    useWorkspace,
+} from "@gooddata/sdk-ui";
+import invariant from "ts-invariant";
+
+/**
+ * @beta
+ */
+export interface IUseDeleteWidgetAlertConfig extends UseCancelablePromiseCallbacks<void, GoodDataSdkError> {
+    /**
+     * Widget alert to delete.
+     */
+    widgetAlert?: IWidgetAlert;
+
+    /**
+     * Backend to work with.
+     *
+     * Note: the backend must come either from this property or from BackendContext. If you do not specify
+     * backend here, then the hook MUST be called within an existing BackendContext.
+     */
+    backend?: IAnalyticalBackend;
+
+    /**
+     * Workspace to work with.
+     *
+     * Note: the workspace must come either from this property or from WorkspaceContext. If you do not specify
+     * workspace here, then the hook MUST be called within an existing WorkspaceContext.
+     */
+    workspace?: string;
+}
+
+/**
+ * Hook allowing to delete a widget alert
+ * @param config - configuration of the hook
+ * @beta
+ */
+export function useDeleteWidgetAlert({
+    widgetAlert,
+    backend,
+    workspace,
+    onCancel,
+    onError,
+    onLoading,
+    onPending,
+    onSuccess,
+}: IUseDeleteWidgetAlertConfig): UseCancelablePromiseState<void, any> {
+    const effectiveBackend = useBackend(backend);
+    const effectiveWorkspace = useWorkspace(workspace);
+
+    invariant(
+        effectiveBackend,
+        "The backend in useDeleteKpiAlert must be defined. Either pass it as a config prop or make sure there is a BackendProvider up the component tree.",
+    );
+
+    invariant(
+        effectiveWorkspace,
+        "The workspace in useDeleteKpiAlert must be defined. Either pass it as a config prop or make sure there is a WorkspaceProvider up the component tree.",
+    );
+
+    const promise = widgetAlert
```

Comment:
I can be wrong, but because useCancelablePromise is not really canceling the promise (they are not really cancelable - useCancelablePromise just don't care about the results anymore...), usage for delete/update operations can be problematic, because first canceled promise can finish the operation, and second one can throw backend error (as it is already updated/deleted). And we don't know which request will be resolved first. What do you think?

Reply:
Hm... that's a good point, I just used it for convenience, but you are right the cancellable does not make much sense here. I will try to avoid it

Reply:
Descoped from this, I will create a follow-up task.